packages/project-disaster-trail/src/components/Game/TaskScreen/index.js

```diff
@@ -1,40 +1,240 @@
-import React, { memo, useEffect } from "react";
+import { memo, useEffect, useState, Fragment } from "react";
 import { PropTypes } from "prop-types";
 import { connect } from "react-redux";
+/** @jsx jsx */
+import { css, jsx } from "@emotion/core";
 
 import { goToNextChapter } from "../../../state/chapters";
-import { getActiveTaskData } from "../../../state/tasks";
+import {
+  getActiveTaskData,
+  completeTask,
+  getWeightedTasks
+} from "../../../state/tasks";
+import { getPlayerKitItems } from "../../../state/kit";
+import OrbManager from "../OrbManager";
 import ChooseScreen from "./ChooseScreen";
 import SolveScreen from "./SolveScreen";
+import DurationBar from "../../atoms/DurationBar";
+import Ticker from "../../atoms/Ticker";
+import TaskDebugger from "../../atoms/TaskDebugger";
+import Timer from "../../../utils/timer";
 
-const TaskScreen = ({ activeTask, endChapter }) => {
-  // Prevent calling useEffect again with 2nd argument so timer isn't deleted
+import * as ACTIONS from "../../../constants/actions";
+import usePrevious from "../../../state/hooks/usePrevious";
+
+const mapAndInfoStyle = css`
+  position: relative;
+  display: grid;
+  overflow: hidden;
+  width: 100%;
+  height: 100%;
+  grid-template-columns: 1fr;
+  background: beige;
+`;
+
+const TaskScreen = ({
+  activeTask,
+  completeActiveTask,
+  endChapter,
+  debug = true,
+  playerKitItems,
+  weightedTasks
+}) => {
+  const prevActiveTask = usePrevious(activeTask);
+  const [action, setAction] = useState(ACTIONS.SOLVING);
+  const prevAction = usePrevious(action);
+  const [percentComplete, setPercentComplete] = useState(0);
+  const [timer] = useState(new Timer());
+  const [chapterTimer] = useState(new Timer());
+  const [votingComplete, setVotingComplete] = useState(false);
+  const [movingMapComplete, setMovingMapComplete] = useState(false);
+
+  const votingDuration = 20;
+  const mapTransitionDuration = 5;
+  const chapterDuration = 60;
+
+  // 1) Solve Screen
+  // 2) Vote
+  // 3) Move Map
+  // 4) Go to step 1?
+  const onTimerComplete = () => {
+    switch (action) {
+      case ACTIONS.SOLVING:
+        if (activeTask) {
+          completeActiveTask(activeTask.id);
+        }
+        break;
+      case ACTIONS.VOTING:
+        setVotingComplete(true);
+        break;
+      case ACTIONS.MOVING_MAP:
+        setMovingMapComplete(true);
+        break;
+      default:
+        // eslint-disable-next-line no-console
+        console.warn("Unknown action in onTimerComplete ", action);
+        break;
+    }
+  };
+
+  const startTimer = duration => {
+    // const d = debug ? 5 : duration;
+    timer.setDuration(duration);
+    timer.reset();
+    timer.addCallback((t, p) => {
+      setPercentComplete(p);
+    });
+    timer.addCompleteCallback(() => onTimerComplete());
+    timer.start();
+  };
+
+  // when the component mounts, start a timer of the active task's time
```

Comment:
I think it would be helpful to have one-line comments like this above all the `useEffect`s so that it's easy for other developers to parse what's going on

Reply:
Will do